### PR TITLE
Raise error if missing artwork location in ShippingHelper.calculate

### PR DESCRIPTION
**Pull request by williardx**
### Problem
In rare instances it's possible to end up attempting to calculate shipping with a missing artwork location.

Addresses [PURCHASE-786](https://artsyproduct.atlassian.net/browse/PURCHASE-786).

### Solution
Raise a `missing_artwork_location` error if the artwork location is blank when we try to calculate shipping costs.

### What changed
- `ShippingHelper#calculate` raises an error if artwork location is blank when trying to calculate shipping costs
- Refactored domestic shipping logic in `calculate` into its own method `domestic?`

### CODE REVIEW

File: lib/shipping_helper.rb [hunk 1]
```diff
@@ -1,11 +1,24 @@
 class ShippingHelper
-  def self.calculate(artwork, fulfillment_type, shipping_address = nil)
-    return 0 if fulfillment_type == Order::PICKUP
+  def initialize(artwork, fulfillment_type, shipping_address = nil)
```
[the remark this answers is not in the document]
Comment (williardx) [reply lib/shipping_helper.rb]: That makes sense to me. The refactor was actually prompted by Rubocop complaining about the complexity of `calculate!`, which I think is fair. How do you see helper methods working for these single-method classes? Would those just be a private class method?

[the remark this answers is not in the document]
Comment (ashkan18) [reply lib/shipping_helper.rb]: yes we can have things like `domestic?` to be a private class method. Those are usually little awkward in my opinion in Ruby but they are totally fine to have

[the remark this answers is not in the document]
Comment (williardx) [reply lib/shipping_helper.rb]: Cool -- I'll change it back, then, and use a private class method for `domestic?`

**Comment by ashkan18** [comment]
🙌 thanks for quick fix

**Pull request merged**